docs/docs/data-storage-redux.md

```diff
@@ -1,9 +1,51 @@
 ---
 title: Data Storage (Redux)
-issue: https://github.com/gatsbyjs/gatsby/issues/21068
 ---
 
-This is a stub. Help our community expand it.
+During Gatsby's bootstrap & build phases, the state is stored and manipulated using the [Redux](https://redux.js.org/) library. The key purpose of using Redux in Gatsby internals is to centralize all of the state logic. Reviewing the Gatsby [reducers](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby/src/redux/reducers) and [actions](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby/src/redux/actions) folders gives a comprehensive picture of what state manipulations are possible.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Store
+
+The namespaces in Gatsby's Redux store are a great overview of the Gatsby internals, here are a few:
+
+- **[Nodes](/docs/node-interface/)** - All data that’s added to Gatsby is modeled using nodes. Nodes are most commonly added by Source plugins such as `gatsby-source-filesystem`.
+- **[Schema](/docs/schema-generation/)** - GraphQL schema inferred from Nodes, available for querying by page and static queries.
+- **[Pages](/docs/gatsby-internals-terminology/#page)** - A `Map` of page paths to page objects. Page objects contain information needed to render a page such as component file path, page query and context.
+- **[Components](/docs/gatsby-internals-terminology/#component)** - A `Map` of component file paths to page objects.
+- **[Static Query Components](/docs/static-vs-normal-queries/#keeping-track-of-site-queries-during-build-in-redux-stores)** - A `Map` of any components detected with a static query.
+- **Jobs** - Long-running processes, generally started as a side-effect to a GraphQL query. Gatsby doesn’t finish its process until all jobs are ended.
+- **Webpack** - Config for the [Webpack](https://webpack.js.org/) bundler which handles code optimization and splitting of delivered javascript bundles.
+
+The Gatsby [/redux index file](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby/src/redux/index.ts) has two key exports, `store` and `emitter`. Throughout the bootstrap and build phases, `store` is used to get the current state and dispatch actions, while `emitter` is used to register listeners for particular actions. The store is also made available to Gatsby users through the [Node APIs](/docs/node-apis/).
```

Comment:
```suggestion
The Gatsby [redux index file](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby/src/redux/index.ts) has two key exports, `store` and `emitter`. Throughout the bootstrap and build phases, `store` is used to get the current state and dispatch actions, while `emitter` is used to register listeners for particular actions. The store is also made available to Gatsby users through the [Node APIs](/docs/node-apis/).
```
The slash in `/redux` reads a bit strange in a sentence, I'd recommend omitting it or marking it as code.